common/src/main/scala/explore/components/undo/Restorer.scala

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.components.undo
+
+import cats.effect.IO
+import monocle.Getter
+
+// We don't use a case class to avoid the type parameter on T
+trait Restorer[M] { // M = (Local) Model
+  type T // T = Value type
+
+  val value: T // Value that will be restored upon undo/redo
+  val getter: Getter[
+    M,
+    T
+  ] // How to refresh the value from the model. Used when going from undo=>redo or viceversa.
+  val setter: T => IO[Unit] // Modify the model
```

Comment:
Should we parametrize `IO`?

Reply:
Great idea. I'm doing it in a bunch of other places too (`Undoer` backend, `SubscriptionRender`, `SubscriptionRenderMod`).